### ISSUE TITLE
False positives missing arg on inherited class

### ISSUE THREAD

**Issue by ion-elgreco** [astral-sh/ty#310]
### Summary

Example:

```python
import dagster as dg

class Foo(dg.ConfigurableResource):
    def do_something(self, first_arg: str) -> None:
        print(first_arg)
        return None

Foo().do_something("random")
```

Returns:
```python
error[missing-argument]: No argument provided for required parameter `self` of function `__init__`
  --> test.py:10:1
   |
10 | Foo().do_something("random")
   | ^^^^^
   |
info: `missing-argument` is enabled by default

error[missing-argument]: No argument provided for required parameter `first_arg` of function `do_something`
  --> test.py:10:1
   |
10 | Foo().do_something("random")
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
info: `missing-argument` is enabled by default
```

### Version

ty 0.0.0-alpha.8 (0474b40e1 2025-05-09)

**Comment by sharkdp** [comment]
I think the underlying cause is the same as #312. I'm going to close this as a duplicate. Thank you for reporting it.
